src/FlexItem.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import cx from 'classnames';
+
+export const FLEX_ITEM_CLASS = 'flex-item';
+export const FLEX_ITEM_SHRINK_CLASS = 'flex-item--shrink';
+export const FLEX_ITEM_GROW_CLASS = 'flex-item--';
+export const FLEX_GROW_FACTORS = [1,2,3,4,5,6,7];
+/**
+ * Design System Component: Provides `FlexItem` styled container for use in `Flex` component containers
+ * @module FlexItem
+ */
+class FlexItem extends React.Component {
+	/**
+	 * @return {React.element} the commend form React element
+	 */
+	render() {
+		const {
+			children,
+			className,
+			shrink,
+			growFactor,
+			...other
+		} = this.props;
+
+		const classNames = cx(
+			FLEX_ITEM_CLASS,
+			{
+				[FLEX_ITEM_SHRINK_CLASS]: shrink,
+				[`${FLEX_ITEM_GROW_CLASS}${growFactor}`] : growFactor,
+			},
+			className
```

Comment:
Another nitpick: I am not a fan of the large object block in the middle of the classNames function. I prefer something more compact like:

```js
const classNames = cx(
   FLEX_ITEM_CLASS,
   { FLEX_ITEM_SHRINK_CLASS: shrink },
   { [`${FLEX_ITEM_GROW_CLASS}${growFactor}`]: growFactor },
   className
);
```
This is a personal preference. How you have it is completely fine as well.